### PR TITLE
Slight cleanup of `cargo_build_script_runner` + ran rustfmt

### PR DESCRIPTION
**Pull request by UebelAndre**
I was trying to read through this code and found it a bit hard to follow. This (to me) makes it a bit more readable. There should otherwise be no functional change here.

### CODE REVIEW

**Comment by UebelAndre** [comment]
@illicitonion Hey, if you're around this weekend, I'd love a review. I think this one should be quick.

If you don't think this code should be touched then I'm happy to have this PR rejected. I get the sense that I'm touching some sensitive code (but maybe that's not well founded?).

File: cargo/cargo_build_script_runner/bin.rs [hunk 1]
```diff
@@ -31,118 +31,184 @@ fn main() -> Result<(), String> {
...
+            Some(progname), 
+            Some(crate_name), 
+            Some(crate_links), 
+            Some(out_dir), 
+            Some(envfile), 
+            Some(flagfile), 
+            Some(linkflags), 
+            Some(depenvfile)
+        ) => {
+            let mut remaining_args = Vec::new();
```
Comment (illicitonion) [inline cargo/cargo_build_script_runner/bin.rs:196]: Or just `let remaining_args = args.collect();`?

File: cargo/cargo_build_script_runner/bin.rs [hunk 2]
```diff
@@ -31,118 +31,184 @@ fn main() -> Result<(), String> {
...
+struct Options {
+    progname: String,
+    crate_name: String,
+    crate_links: String,
+    out_dir: String,
+    envfile: String,
+    flagfile: String,
+    linkflags: String,
+    depenvfile: String,
+    remaining_args: Vec<String>,
```
Comment (illicitonion) [inline cargo/cargo_build_script_runner/bin.rs:177]: Maybe `input_dep_env_paths` to be more descriptive of what it contains? And maybe call `depenvfile` instead `output_dep_env_path` to differentiate?

**Review by illicitonion — COMMENTED** [review]
Looks good, thanks! Left a couple of minor suggestions.

For the future, if you're `rustfmt`ing things and also changing them, it can be nice to do those in two separate commits, to make it easier to review what _you_ have changed, vs what rustfmt has changed :)

> I get the sense that I'm touching some sensitive code (but maybe that's not well founded?).

I don't think it's particularly sensitive - the only thing to know about it is that it currently can't have dependencies, which is why some of it is how it is (e.g. the command line parsing) :)

File: cargo/cargo_build_script_runner/lib.rs [hunk 3]
```diff
@@ -144,7 +144,11 @@ impl BuildScriptOutput {
         v.iter()
             .filter_map(|x| {
                 if let BuildScriptOutput::DepEnv(env) = x {
-                    Some(format!("{}{}", prefix, Self::redact_exec_root(env, exec_root)))
+                    Some(format!(
```
Comment (UebelAndre) [inline cargo/cargo_build_script_runner/lib.rs:147]: > For the future, if you're `rustfmt`ing things and also changing them, it can be nice to do those in two separate commits, to make it easier to review what _you_ have changed, vs what rustfmt has changed :)

Fortunately, this is the only thing that changed as a result of running `rustfmt`. But I can definitely keep the changes separated in future PRs 👍

**Comment by UebelAndre** [comment]
@illicitonion Ready for another review 😄

File: cargo/cargo_build_script_runner/bin.rs [hunk 4]
```diff
@@ -31,121 +31,182 @@ fn main() -> Result<(), String> {
...
+            Ok(Options{
+                progname,
+                crate_name,
+                crate_links,
+                out_dir,
+                envfile,
+                flagfile,
+                linkflags,
+                output_dep_env_path,
+                input_dep_env_paths: args.collect();
```
Comment (illicitonion) [inline cargo/cargo_build_script_runner/bin.rs:205]:
```suggestion
                input_dep_env_paths: args.collect(),
```

**Review by illicitonion — APPROVED** [review]
Looks good - one tiny typo caught by CI :) Thanks!

File: cargo/cargo_build_script_runner/bin.rs (hunk 4 above)
Comment (UebelAndre) [reply cargo/cargo_build_script_runner/bin.rs]: woah, how did the tests pass? 😓 
```
//cargo/cargo_build_script_runner:test                          (cached) PASSED in 0.2s
//test:greeting_rule_test                                       (cached) PASSED in 0.3s
//test:hello_lib_rule_test                                      (cached) PASSED in 0.1s
//test:hello_world_rule_test                                    (cached) PASSED in 0.4s
//test/build_env:arbitrary_env_test                             (cached) PASSED in 0.4s
//test/build_env:conflicting_deps_test                          (cached) PASSED in 0.2s
//test/chained_direct_deps:mod1_test                            (cached) PASSED in 0.2s
//test/chained_direct_deps:mod2_test                            (cached) PASSED in 0.0s
//test/chained_direct_deps:mod3_doc_test                        (cached) PASSED in 0.7s
//test/chained_direct_deps:mod3_test                            (cached) PASSED in 0.4s
//test/clippy:bad_test                                          (cached) PASSED in 0.4s
//test/clippy:ok_test                                           (cached) PASSED in 0.3s
//test/conflicting_deps:conflicting_deps_test                   (cached) PASSED in 0.3s
//test/inline_test_with_deps/test:inline_test                   (cached) PASSED in 0.5s
//test/inline_test_with_deps/test_with_srcs:inline_test         (cached) PASSED in 0.4s
//test/process_wrapper:arg_files_test                           (cached) PASSED in 0.5s
//test/process_wrapper:basic_test                               (cached) PASSED in 0.1s
//test/process_wrapper:combined_test                            (cached) PASSED in 0.5s
//test/process_wrapper:copy_output_test                         (cached) PASSED in 0.1s
//test/process_wrapper:env_files_test                           (cached) PASSED in 0.1s
//test/process_wrapper:stderr_test                              (cached) PASSED in 0.2s
//test/process_wrapper:stdout_test                              (cached) PASSED in 0.2s
//test/process_wrapper:subst_pwd_test                           (cached) PASSED in 0.1s
//test/renamed_deps:mod1_test                                   (cached) PASSED in 0.4s
//test/renamed_deps:mod2_test                                   (cached) PASSED in 0.4s
//test/renamed_deps:mod3_test                                   (cached) PASSED in 0.3s
//test/rust_test_binary:wrapped_rust_test                       (cached) PASSED in 0.2s
//test/rustfmt:rustfmt_test                                     (cached) PASSED in 0.4s
//test/test_env:test                                            (cached) PASSED in 0.6s
//tools/runfiles:runfiles_doc_test                              (cached) PASSED in 0.7s
//tools/runfiles:runfiles_test                                  (cached) PASSED in 0.5s
```

This has been updated.

Comment (UebelAndre) [reply cargo/cargo_build_script_runner/bin.rs]: I guess because that file is not covered by these tests.

**Comment by UebelAndre** [comment]
@illicitonion CI is green. Sorry about that 😅

File: cargo/cargo_build_script_runner/bin.rs (hunk 4 above)
Comment (illicitonion) [reply cargo/cargo_build_script_runner/bin.rs]: Yeah, these don't depend on any build scripts :)
